### PR TITLE
adds CloudFront module for the PublicSuffix site

### PR DESCRIPTION
**Pull request by danielhartnell**
Submitting my work in progress.

### CODE REVIEW

File: nubis/terraform/sites/main.tf [hunk 1]
```diff
@@ -7,6 +7,18 @@ module "consul" {
...
 }
 
+module "cloudfront" {
+  source                 = "github.com/nubisproject/nubis-terraform//cloudfront?ref=685e9714ec80afb3c42ee3ce6b7f641b65f33349"
+  enabled                = "${var.enable_cdn}"
+  region                 = "${var.region}"
+  environment            = "${var.environment}"
+  account                = "${var.account}"
+  service_name           = "${var.service_name}"
+  load_balancer_web      = "${module.load_balancer_web.address}"
```
Comment (danielhartnell) [inline nubis/terraform/sites/main.tf]: I'm not sure if the `load_balancer_web` module output is actually in scope here. I might have to refer to that elsewhere.

**Comment by danielhartnell** [comment]
I also don't love the idea of assuming that the first element in the `domain_aliases` list has same name as the ACM certificate. I might try to find a better way to handle that.

**Comment by danielhartnell** [comment]
I have tested this as well as I can locally. Without having a copy of Haul deployed to a test account, I do still have some unknowns. I wasn't able to test and see if the `load_balancer_web` output is what I expect (ELB address to use as the origin).

Before merging this, I need someone to make sure that an ACM certificate exists in `us-east-1` for the following domains: `["publicsuffix.org", "www.publicsuffix.org"]`.

**Comment by gozer** [comment]
Can't you generate the ACM certs from Terraform instead ?

Using https://www.terraform.io/docs/providers/aws/r/acm_certificate.html

Just trying to avoid manual processes

**Comment by danielhartnell** [comment]
@gozer,

I would like to spend some more time thinking about how to setup ACM rather than quickly adding it right now. I suspect the Terraform should probably go into the CloudFront module as well. I created an issue to follow up on that:

https://github.com/nubisproject/nubis-terraform/issues/209

In the meantime, it'll help to get that ACM cert for publicsuffix.org in us-east-1.

**Pull request merged**